tests/tests/dispatch_workgroups_indirect.rs

```diff
@@ -0,0 +1,162 @@
+                out[1] = num_workgroups.y;
+                out[2] = num_workgroups.z;
+            }
+        }
+    ";
+
+    let module = ctx
+        .device
+        .create_shader_module(wgpu::ShaderModuleDescriptor {
+            label: None,
+            source: wgpu::ShaderSource::Wgsl(SHADER_SRC.into()),
+        });
+
+    let pipeline = ctx
+        .device
+        .create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+            label: None,
+            layout: None,
+            module: &module,
+            entry_point: "main",
+            compilation_options: Default::default(),
+            cache: None,
+        });
+
+    let out_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+        label: None,
+        size: 12,
+        usage: wgpu::BufferUsages::STORAGE | wgpu::BufferUsages::COPY_SRC,
+        mapped_at_creation: false,
+    });
+
+    let readback_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+        label: None,
+        size: 12,
+        usage: wgpu::BufferUsages::COPY_DST | wgpu::BufferUsages::MAP_READ,
+        mapped_at_creation: false,
+    });
+
+    const INDIRECT_OFFSET: u64 = 256 + 4; // unaligned offset (min_storage_buffer_offset_alignment is 256 by default)
+
+    let indirect_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+        label: None,
+        size: 12 + INDIRECT_OFFSET + 8, // indirect args + offset + extra space (to make sure our binding code works properly)
+        usage: wgpu::BufferUsages::COPY_DST | wgpu::BufferUsages::INDIRECT,
+        mapped_at_creation: false,
+    });
+
+    ctx.queue.write_buffer(
+        &indirect_buffer,
+        INDIRECT_OFFSET,
+        bytemuck::bytes_of(num_workgroups),
+    );
+
+    let bind_group = ctx.device.create_bind_group(&wgpu::BindGroupDescriptor {
+        label: None,
+        layout: &pipeline.get_bind_group_layout(0),
+        entries: &[wgpu::BindGroupEntry {
+            binding: 0,
+            resource: out_buffer.as_entire_binding(),
+        }],
+    });
+
+    let mut encoder = ctx
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+    {
+        let mut compute_pass = encoder.begin_compute_pass(&wgpu::ComputePassDescriptor::default());
+        compute_pass.set_pipeline(&pipeline);
+        if !forget_to_set_bind_group {
+            compute_pass.set_bind_group(0, &bind_group, &[]);
+        }
+        compute_pass.dispatch_workgroups_indirect(&indirect_buffer, INDIRECT_OFFSET);
+    }
+
+    encoder.copy_buffer_to_buffer(&out_buffer, 0, &readback_buffer, 0, 12);
+
+    ctx.queue.submit(Some(encoder.finish()));
+
+    readback_buffer
+        .slice(..)
+        .map_async(wgpu::MapMode::Read, |_| {});
+
+    ctx.async_poll(wgpu::Maintain::wait())
+        .await
+        .panic_on_timeout();
+
+    let view = readback_buffer.slice(..).get_mapped_range();
+
+    bytemuck::from_bytes::<[u32; 3]>(&view).to_owned()
```

Comment:
This seems simpler, since `[u8; N]` is `Copy`, and we can let type inference supply the type parameter:

```suggestion
    *bytemuck::from_bytes(&view)
```